tempodb/tempodb.go

```diff
@@ -246,14 +246,28 @@ func (rw *readerWriter) Find(ctx context.Context, tenantID string, id encoding.I
 	rw.blockListsMtx.Lock()
 	blocklist, found := rw.blockLists[tenantID]
 	copiedBlocklist := make([]interface{}, 0, len(blocklist))
+
+	var err error
```

Comment:
this code becomes much simpler if take a uuid.UUID directly.  no reason to double check if there's a value, just always respect the values passed in.